app/actions/handlers.py

```diff
@@ -145,9 +142,33 @@ async def action_pull_events(integration:Integration, action_config: PullEventsC
         )
         return [msg]
 
-        # Create a list of tasks.
-    tasklist = [asyncio.create_task(get_fire_alerts(integration, action_config, auth_config, aoi_data)),
-                 asyncio.create_task(get_integrated_alerts(integration, action_config, auth_config))
+    # Get AOI and Geostore data.
+    aoi_id = await dataapi.aoi_from_url(action_config.gfw_share_link_url)
+    aoi_data = await dataapi.get_aoi(aoi_id=aoi_id)
+
+    geostore_ids = await state_manager.get_geostore_ids(aoi_data.id)
+
+    if not geostore_ids:
+        geostore:Geostore = await dataapi.get_geostore(geostore_id=aoi_data.attributes.geostore)
+
+        geometry_collection = GeometryCollection(
+            [
+                shape(feature["geometry"]).buffer(0)
+                for feature in geostore.attributes.geojson["features"]
+            ]
+        )
+        for partition in utils.generate_geometry_fragments(geometry_collection=geometry_collection):
+
+            geostore = await dataapi.create_geostore(geometry=mapping(partition))
+
+            await state_manager.add_geostore_id(aoi_data.id, geostore.gfw_geostore_id)
+
+        await state_manager.set_geostores_id_ttl(aoi_data.id, 86400*7)
+
+    # Create a list of tasks.
+    sema = asyncio.Semaphore(app.settings.GFW_DATASET_QUERY_CONCURRENCY)
+    tasklist = [asyncio.create_task(get_nasa_viirs_fire_alerts(integration, action_config, auth_config, aoi_data, sema)),
+                 asyncio.create_task(get_integrated_alerts(integration, action_config, auth_config, aoi_data, sema))
```

Comment:
`create_task()` seems to be redundant here since we do `await gather()` right after. Keep in mind that `create_task` also schedules the coroutines to run.

Reply:
Thanks. I'll fix that now.